cookbook/helper/ingredient_parser.py

```diff
@@ -249,15 +262,31 @@ def apply_never_unit_automations(self, tokens):
 
         return tokens
 
-    def parse_tokens(self, tokens):
+    def apply_transpose_words_automations(self, ingredient):
         """
-        parser that applies automations to unmodified tokens
+        If two words (param_1 & param_2) are detected in sequence, swap their position in the ingredient string
+        :param 1: first word to detect
+        :param 2: second word to detect
+        return: new ingredient string
         """
 
+        ####################################################
+        ####################################################
+        ####################################################
+        ####################################################
         if self.ignore_rules:
-            return tokens
+            return ingredient
+
+        else:
+            if self.transpose_words:
+                for rule in self.transpose_words:
+                    ingredient = re.sub(rf"\b({rule[0]}) ({rule[1]})\b", r"\2 \1", ingredient)
+
+            else:
+                for rule in Automation.objects.filter(space=self.request.space, type=Automation.TRANSPOSE_WORDS, disabled=False).order_by('order'):
```

Comment:
this hits the database for every ingredient parsed and might also execute extremly large amounts of regex

Reply:
That would only occur if not cached - I can take that out and have it so the rule never runs if the cache doesn't exist.  Or do you thoughts on another way to limit it?